fedbiomed/researcher/cli.py

```diff
@@ -13,16 +13,16 @@
 
 
 __intro__ = """
-   __         _ _     _                          _ 
+   __         _ _     _                          _
   / _|       | | |   (_)                        | |
  | |_ ___  __| | |__  _  ___  _ __ ___   ___  __| |
  |  _/ _ \/ _` | '_ \| |/ _ \| '_ ` _ \ / _ \/ _` |
- | ||  __/ (_| | |_) | | (_) | | | | | |  __/ (_| |    _ 
- |_| \___|\__,_|_.__/|_|\___/|_| |_| |_|\___|\__,_|   | |              
-                     _ __ ___  ___  ___  __ _ _ __ ___| |__   ___ _ __ 
+ | ||  __/ (_| | |_) | | (_) | | | | | |  __/ (_| |    _
+ |_| \___|\__,_|_.__/|_|\___/|_| |_| |_|\___|\__,_|   | |
+                     _ __ ___  ___  ___  __ _ _ __ ___| |__   ___ _ __
                     | '__/ _ \/ __|/ _ \/ _` | '__/ __| '_ \ / _ \ '__|
-                    | | |  __/\__ \  __/ (_| | | | (__| | | |  __/ |   
-                    |_|  \___||___/\___|\__,_|_|  \___|_| |_|\___|_|   
+                    | | |  __/\__ \  __/ (_| | | | (__| | | |  __/ |
+                    |_|  \___||___/\___|\__,_|_|  \___|_| |_|\___|_|
```

Comment:
@srcansiz we have an issue when using alternate researcher config file name for launching researcher eg

```
# creates a config file config-researcher-alternate.ini and also a config_researcher.ini
/scripts/fedbiomed_run researcher --config config-researcher-alternate.ini start
```

Then in the notebook
```
from fedbiomed.researcher.config import config
# config.read() 
config._cfg.__dict__
```

This is the `config_researcher.ini` which is read not the `config-researcher-alternate.ini`

We may fix this in `fedbiomed.researcher.cli`:

https://github.com/fedbiomed/fedbiomed/blob/e7babbb3de24717e0d6e7ef4fbbde1e3d540d40d/fedbiomed/researcher/cli.py#L56

With something like (looks to work for me):

```
        import os
        config_file = os.environ.get("CONFIG_FILE")
        if config_file:
            os.environ["FBM_RESEARCHER_CONFIG_FILE"] = config_file

        # apply change above before creating a subprocess for the notebook
        command = ["jupyter", "notebook"]
        command = [*command, *options]
        process = subprocess.Popen(command)
```

This enables using the proper config file  `config-researcher-alternate.ini` (please double check my quick test).

**Yet we also need to fix the (unwanted, duplicate)** creation of `config_researcher.ini` in this case (did not investigate this aspect).

Reply:
hello @srcansiz it looks like the following would solve:
- passing the alternate file name to the researcher
- not creating a researcher config file with the default file name

It is a replacement for the modifications proposed above.

Would you give an eye/test and give some feedback (or implement if it seems ok) ?

In `fedbiomed.researcher.cli` in `ConfigNameActionResearcher`, replace the `set_component` method with

```
            def set_component(self, config_name: str) -> None:
                # dynamically creates the config after the config file name was set
                config_file = os.environ.get("CONFIG_FILE")
                if config_file:
                    os.environ["FBM_RESEARCHER_CONFIG_FILE"] = config_file

                importlib.import_module("fedbiomed.researcher.config")
```

1. removes the dependency to `environ` (thus config file creation with default name)
2. creates a config file with given name (it is done after getting the config name, as we import dynamically)
3. ensures passing the researcher config file name in the notebook

Reply:
Hi @mvesin,

Thanks for pointing this issues and the proposal. I implemented as you suggested.